### PR TITLE
Added documentation to cover the different build variants of Commons and how to switch between them.

### PR DESCRIPTION
**Pull request by Glitch101**
This PR is the result of the [issue](https://github.com/commons-app/apps-android-commons/issues/3409) that was created in the main Commons repo. When the beta servers were down, most of us weren't able to login to beta servers and as we were new to the Commons project, this confused me and other new contributors on how to change the build variant to use the production server for testing.
This PR fixes that!
It contains the steps and information that were posted on the above-linked issue by the contributors and some more information that I found useful.

### CODE REVIEW

**Comment by domdomegg** [comment]
Thanks for adding these 🎉

File: android/build-variants/Build-Variants.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+`betaDebug`: This version connects to the [beta server](https://commons.wikimedia.beta.wmflabs.org/wiki/Main_Page) and can be used for debugging the app during development.
+
+`prodDebug`: This version connects to the [production server](https://commons.wikimedia.org/wiki/Main_Page) and can also be used for debugging the app during development.
+
+`betaRelease`: This is the release version of the app which connects to the beta server. This cannot be used for debugging the app. You would have to [sign this version of the app](https://developer.android.com/studio/publish/app-signing) using your signing keys.
+
+`prodRelease`: This is the release version of the app which connects to the production server. This cannot be used for debugging the app. You would have to sign this version of the app using your signing keys.
+
+For information on how to change `build variants`, [this page can be referred](https://github.com/Glitch101/documentation/blob/master/android/build-variants/Changing-Build-Variants.md).
```
Comment (maskaravivek) [inline android/build-variants/Build-Variants.md]: @Glitch101 Can you update this link and use a page that is part of this repo.

Comment (Glitch101) [reply android/build-variants/Build-Variants.md]: Thanks for saving this blunder. I was just about to delete my fork. I have made the fixes in my fork but this PR has already been merged.

Comment (Glitch101) [reply android/build-variants/Build-Variants.md]: Ah, I have opened [another PR](https://github.com/commons-app/documentation/pull/2) to fix the links. Thanks!

**Comment by Glitch101** [comment]
> Thanks for adding these 🎉

My pleasure :)

**Pull request merged**